examples/sse/src/main.rs

```diff
@@ -44,16 +43,35 @@ async fn main() {
     axum::serve(listener, app).await.unwrap();
 }
 
+fn app() -> Router {
+    let assets_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("assets");
+    let static_files_service = ServeDir::new(assets_dir).append_index_html_on_directories(true);
+    // build our application with a route
+    Router::new()
+        .fallback_service(static_files_service)
+        .route("/sse", get(sse_handler))
+        .layer(TraceLayer::new_for_http())
+}
+
+#[derive(Deserialize)]
+struct Repeat {
+    take_n: Option<usize>,
+}
+
 async fn sse_handler(
     TypedHeader(user_agent): TypedHeader<headers::UserAgent>,
+    repeat: Query<Repeat>,
```

Comment:
Adding this specifically for testing feels wrong. I think the test should be able to just take a certain number of events itself. Perhaps using https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.take

Reply:
It's using `.take` at line 74, maybe the question is how do you recommend to pass n to `.take()` in the test?

Reply:
Like this? https://github.com/tokio-rs/axum/pull/2465/commits/9cc61f1318fc65b880b2bcc682cab98e2e6eb838

Reply:
Yes!